tests/src/tools/ReadyToRun.SuperIlc/BuildFolderSet.cs

```diff
@@ -151,26 +150,97 @@ public bool Compile()
                 }
             }
 
-            _logWriter.WriteLine($"Compiled {successfulCompileCount} / {successfulCompileCount + failedCompilationsPerBuilder.Count} assemblies in {stopwatch.ElapsedMilliseconds} msecs.");
+            _compilationMilliseconds = stopwatch.ElapsedMilliseconds;
+
+            return success;
+        }
+
+        public bool CompileFramework()
```

Comment:
This seems to be something that the compile-directory command could do on our behalf, either in an initial separate invocation of SuperIlc, before building a folder set, or by invoking the compile-directory command with the options configured appropriately. I think that would help simplify the command line since if we do SuperIlc compile-directory, then SuperIlc Build-Subtree, we can just pass in the pre-compiled framework as a regular reference folder files can be copied from into the output. My initial goal for SuperIlc is we'd have a composable set of blocks we can mix and match to easily create complex emergent scenarios.

Reply:
Well, that's for sure. I'm however still convinced that it is generally useful to have a single command line invocation cover most common test scenarios.

Reply:
Well, SuperIlc currently supports both; you can either do the simple thing i.e. use --framework to build and use the CPAOT / Crossgen-compiled framework in your build; or you can prebuild it manually using the compile-directory command and subsequently just consume it using the --use-framework option.

While I understand your building block concept, the app source code itself is also internally built from building blocks and the extra functionality seems both trivial to implement and somewhat easier to use in some scenarios as you don't need any additional scripting to do testing end-to-end. Would you mind temporarily keeping this functionality before we get more practical experience with actual usage of the tool?

Reply:
Sure I'm fine with the changes as-is